source/extensions/quic_listeners/quiche/envoy_quic_server_session.cc

```diff
@@ -0,0 +1,181 @@
+#include "extensions/quic_listeners/quiche/envoy_quic_server_session.h"
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+
+#include "quiche/quic/core/quic_crypto_server_stream.h"
+#pragma GCC diagnostic pop
+
+#include "extensions/quic_listeners/quiche/envoy_quic_server_stream.h"
+
+namespace Envoy {
+namespace Quic {
+
+EnvoyQuicServerSession::EnvoyQuicServerSession(
+    const quic::QuicConfig& config, const quic::ParsedQuicVersionVector& supported_versions,
+    quic::QuicConnection* connection, quic::QuicSession::Visitor* visitor,
+    quic::QuicCryptoServerStream::Helper* helper, const quic::QuicCryptoServerConfig* crypto_config,
+    quic::QuicCompressedCertsCache* compressed_certs_cache, Event::Dispatcher& dispatcher)
+    : quic::QuicServerSessionBase(config, supported_versions, connection, visitor, helper,
+                                  crypto_config, compressed_certs_cache),
+      filter_manager_(*this), dispatcher_(dispatcher), stream_info_(dispatcher.timeSource()) {}
+
+EnvoyQuicServerSession::~EnvoyQuicServerSession() { delete connection(); }
```

Comment:
nit: add a member variable to own the connection, and remove the delete: `std::unique_ptr<quic::QuicConnection> envoy_quic_connection_;`

I think we can change the constructor's `quic::QuicConnection*` to a `EnvoyQuicConnection*` (and own it in a `std::unique_ptr<EnvoyQuicConnection>`), otherwise this class may need to down_cast to call methods specific to EnvoyQuicConnection.

Reply:
good idea! done